### PR TITLE
Add README.md to beats packages

### PR DESCRIPTION
**Pull request by ruflin**
* Add README.md to homedir to be part of each package
* Update description of each beat to make more sense in this context
* Add version.yml to package.yml file so it doesn't have to be added in each build

Closes https://github.com/elastic/beats/issues/1064

### CODE REVIEW

File: dev-tools/packer/readme.md.j2 [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+{{.beat_description}}
+
+## Getting Started
+
+To get running with your beat, setup elasticsearch on your localhost and start it with:
+
+     ./{{.beat_name}}  -c {{.beat_name}}.yml -e
+
+This will start the beat and send to do your elasticsearch instance. To load the dashboards for {{.beat_name}} into Kibana, run:
```
Comment (andrewkroh) [inline dev-tools/packer/readme.md.j2]: `s/send to do/send to/`

File: packetbeat/Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 
 BEATNAME?=packetbeat
-BEAT_DESCRIPTION?=A network analyzer tool that sends data to Elasticsearch.
+BEAT_DESCRIPTION?=Packetbeat: A network analyzer tool that sends data to Elasticsearch.
```
Comment (andrewkroh) [inline packetbeat/Makefile]: Since it's not consistent with the other descriptions, how about "Packetbeat analyzes network traffic and sends the data to Elasticsearch."

File: dev-tools/packer/readme.md.j2 (hunk 1 above)
Comment (ruflin) [inline dev-tools/packer/readme.md.j2]: I adjusted it to: This will start the beat and send the data to your elasticsearch instance.

File: packetbeat/Makefile (hunk 2 above)
Comment (ruflin) [inline packetbeat/Makefile]: done

**Comment by ruflin** [comment]
jenkins, package it

File: dev-tools/packer/readme.md.j2 [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Welcome to {{.beat_name}} {{.version}}
+
+{{.beat_description}}
+
+## Getting Started
```
Comment (tsg) [inline dev-tools/packer/readme.md.j2]: We should check with @dedemorton how she feels about having this lightweight getting started guide in here, vs just a link to the docs getting started.

File: dev-tools/packer/readme.md.j2 [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+To get running with your beat, setup elasticsearch on your localhost and start it with:
```
Comment (tsg) [inline dev-tools/packer/readme.md.j2]: Capital "Elasticsearch". Also the way it's written know it sounds a bit like that is the command to start ES.

File: dev-tools/packer/readme.md.j2 [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+     ./{{.beat_name}}  -c {{.beat_name}}.yml -e
+
+This will start the beat and send the data to your elasticsearch instance. To load the dashboards for {{.beat_name}} into Kibana, run:
```
Comment (tsg) [inline dev-tools/packer/readme.md.j2]: capitalize "Elasticsearch" please.

File: dev-tools/packer/readme.md.j2 [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+    ./scripts/import_dashboards
+
+For further steps visit the [Getting started](https://www.elastic.co/guide/en/beats/{{.beat_name}}/current/{{.beat_name}}-getting-started.html) guide.
```
Comment (tsg) [inline dev-tools/packer/readme.md.j2]: Here it's tricky, because "current" is not always correct. For example, if you are on version 5.0.0-beta1, the link will point to the 1.3 documentation.

File: dev-tools/packer/readme.md.j2 [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+## Documentation
+
+Visit [Elastic.co Docs](https://www.elastic.co/guide/en/beats/{{.beat_name}}/current/index.html) for the full {{.beat_name}} documentation.
```
Comment (tsg) [inline dev-tools/packer/readme.md.j2]: Same comment as the above, current might be wrong.

File: dev-tools/packer/readme.md.j2 [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+## Release notes
+
+https://www.elastic.co/guide/en/beats/libbeat/current/release-notes-{{.version}}.html
```
Comment (tsg) [inline dev-tools/packer/readme.md.j2]: Also here.

File: dev-tools/packer/platforms/binary/build.sh [hunk 9]
```diff
@@ -8,7 +8,7 @@ ARCHDIR=${BASEDIR}/../../
 # executed from the top directory
 runid=binary-$BEAT-$ARCH
 
-cat ${BUILD_DIR}/package.yml ${ARCHDIR}/archs/$ARCH.yml ${ARCHDIR}/version.yml > ${BUILD_DIR}/settings-$runid.yml
+cat ${BUILD_DIR}/package.yml ${ARCHDIR}/archs/$ARCH.yml > ${BUILD_DIR}/settings-$runid.yml
```
Comment (tsg) [inline dev-tools/packer/platforms/binary/build.sh]: Are these changes required by the PR or just cleanup?

Comment (ruflin) [inline dev-tools/packer/platforms/binary/build.sh]: They are indirectly required as I added version.yml to package.yml as I already need it in before_build.sh. The alternative would be to create an additional .yml file in before_build but as we always need the version, would only add complexity.

**Comment by ruflin** [comment]
About the generation: I would prefer if this file is not generated by make update as otherwise it would probably have to be checked in. 

I wasn't aware that we have the doc-branch in version.asciidoc. Will try to do some magic to get the info from there for the urls.

@dedemorton Before I do additional work, can you comment in general if you see such a file helpful?

File: dev-tools/packer/readme.md.j2 (hunk 3 above)
Comment (dedemorton) [inline dev-tools/packer/readme.md.j2]: I'm fine with this because it points users to the doc. I'm not sure how you will deal with versioning, though.

File: dev-tools/packer/readme.md.j2 (hunk 4 above)
Comment (dedemorton) [inline dev-tools/packer/readme.md.j2]: ..."set up Elasticsearch" ("setup" should be "set up" when you're using it as a verb). :-)

**Comment by dedemorton** [comment]
I think it's a good idea.

File: dev-tools/packer/readme.md.j2 (hunk 5 above)
Comment (ruflin) [inline dev-tools/packer/readme.md.j2]: done

File: dev-tools/packer/readme.md.j2 (hunk 4 above)
Comment (ruflin) [inline dev-tools/packer/readme.md.j2]: changed

**Comment by ruflin** [comment]
The script is now adjusted to fetch the branch info from the docs version. Now all links work except the one for the release notes in master (as there are none). 

@tsg Is this worth backporting to 5.0 or should we start shipping this with 5.1?

**Comment by ruflin** [comment]
jenkins, package it

**Comment by ruflin** [comment]
jenkins, package it

**Comment by ruflin** [comment]
jenkins, retest it

**Comment by ruflin** [comment]
jenkins, package it
